### PR TITLE
Feature/flashing parking spot

### PR DESCRIPTION
**Pull request by Gursaanj**
Parking spots now flash if chosen to be the destination parking spot. Just an indication device to alert players where to park. 

Only issue is the strength of the color is pretty low, so the flashing effect becomes dim and hard to spot sometimes. Possible solutions would include adding a light source prefab around the collider of the parkingSpot prefab and flash that or create a custom material for the parking spot (instead of using a color) and set the emission values to make them more visible. Let me know if I should do that

### CODE REVIEW

**Comment by Gursaanj** [comment]
[Add flashing indication for Target Parking spots](https://trello.com/c/Mt8bUqR5/55-add-flashing-indication-for-target-parking-spots)

**Comment by Gursaanj** [comment]
> Need to answer the questions

Will make a material then and change the emission levels in ParkingDetectionOnUpdate

File: Assets/Script/Game/ParkingDetection.cs [hunk 1]
```diff
@@ -26,17 +43,39 @@ public class ParkingDetection : MonoBehaviour
...
+    private void ParkingDetectionOnUpdate()
+    {
+        if (m_isChosenToBeParkingSpot)
+        {
+            m_tempColor.a = Mathf.Lerp(m_minAlpha, m_maxAlpha, m_lerpInterpolate);
+            m_glowingSpot.GetComponent<SpriteRenderer>().color = m_tempColor;
+
+            m_lerpInterpolate += 0.5f * Time.deltaTime;
+
+            if (m_lerpInterpolate >= 1.0f)
```
[the remark this answers is not in the document]
Comment (Gursaanj) [reply Assets/Script/Game/ParkingDetection.cs]: yup my bad, completely forgot about Pingponging the time

File: Assets/Script/Game/ParkingDetection.cs [hunk 2]
```diff
@@ -26,17 +40,29 @@ public class ParkingDetection : MonoBehaviour
...
     /// <summary>
     /// Chooses the parking spot.
+    /// set the temp Color and alpha so that the glowing spot can be updated accordingly
     /// </summary>
     public void ChooseParkingSpot() 
     {
-        m_glowingSpot.GetComponent<SpriteRenderer>().color = Color.yellow;
-        isChosenToBeParkingSpot = true;
+        m_tempColor = Color.yellow;
+        m_tempColor.a = 0.0f;
```
Comment (m4rtint) [inline Assets/Script/Game/ParkingDetection.cs]: You don't really need the m_tempColor variable anymore. 

m_glowingSpot.GetComponent<SpriteRenderer>().color = Lerp(Color.clear, Color.ChosenColor, PingPongTIme)

Comment (Gursaanj) [reply Assets/Script/Game/ParkingDetection.cs]: Didnt know Color.clear was a thing haha

**Pull request merged**